### PR TITLE
custom vNet and VMSS

### PR DESCRIPTION
**Pull request by shayshahak**
**What this PR does / why we need it**:
* allowing deployment of ACS in a custom vNet with VM Scale Set for Nodes, Availability Set for Masters, CNI and a Private Cluster Endpoint

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: examples/vnet/kubernetesvnet-vmss.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          "privateCluster": {
+              "enabled": true
+          }
+      }
+    },
+    "masterProfile": {
+      "count": 1,
+      "dnsPrefix": "newprivatecluster",
+      "vmSize": "Standard_D2_v2",
+      "vnetSubnetId": "/subscriptions/931980ec-6458-48c2-9668-354453011dac/resourceGroups/RG-NAME/providers/Microsoft.Network/virtualNetworks/VNET-NAME/subnets/SUBNET-NAME",
```
Comment (CecileRobertMichon) [inline examples/vnet/kubernetesvnet-vmss.json]: Let's make the vnetSubnetId `"/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME"` to be consistent with other example files (and remove the hardcoded sub ID)

File: examples/vnet/kubernetesvnet-vmss.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "availabilityProfile": "VirtualMachineScaleSets",
+        "storageProfile": "ManagedDisks"
+      }
+    ],
+    "linuxProfile": {
+      "adminUsername": "azureuser",
+      "ssh": {
+        "publicKeys": [
+          {
+            "keyData": "THE-KEY"
```
Comment (CecileRobertMichon) [inline examples/vnet/kubernetesvnet-vmss.json]: keyData should be left empty

File: examples/vnet/kubernetesvnet-vmss.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      "vnetSubnetId": "/subscriptions/931980ec-6458-48c2-9668-354453011dac/resourceGroups/RG-NAME/providers/Microsoft.Network/virtualNetworks/VNET-NAME/subnets/SUBNET-NAME",
+      "firstConsecutiveStaticIP": "172.16.3.239",
+      "vnetCidr": "172.16.3.0/24"
+    },
+    "agentPoolProfiles": [
+      {
+        "name": "agentpri",
+        "count": 2,
+        "vmSize": "Standard_D2_v2",
+       "vnetSubnetId": "/subscriptions/931980ec-6458-48c2-9668-354453011dac/resourceGroups/RG-NAME/providers/Microsoft.Network/virtualNetworks/VNET-NAME/subnets/SUBNET-NAME",
```
Comment (CecileRobertMichon) [inline examples/vnet/kubernetesvnet-vmss.json]: ditto

**Comment by CecileRobertMichon** [comment]
@shayshahak please add a short description of what the PR does in the PR summary template

File: examples/vnet/kubernetesvnet-vmss.json (hunk 1 above)
Comment (shayshahak) [reply examples/vnet/kubernetesvnet-vmss.json]: done.

File: examples/vnet/kubernetesvnet-vmss.json (hunk 3 above)
Comment (shayshahak) [reply examples/vnet/kubernetesvnet-vmss.json]: done

File: examples/vnet/kubernetesvnet-vmss.json (hunk 2 above)
Comment (shayshahak) [reply examples/vnet/kubernetesvnet-vmss.json]: done

**Comment by CecileRobertMichon** [comment]
@shayshahak you said this PR is "allowing deployment […]" so I would just like to point out that it is already possible to deploy such a cluster, you can define your own apimodel at any time with any supported combination of features. The example apimodels are there to provide guidance on how to define a specific feature in an apimodel and should be a known working feature set.

**Comment by jackfrancis** [comment]
Hi @shayshahak, we're going to close this for the following reasons:

- our example API models are meant to be general purpose, and paired with documentation to provide a working pattern of a particular use case
- we don't want to continue to add to the list, as it may confuse folks coming to the project "which cluster config should I use?"

This area of the project is not clearly documented, and so I'm sorry if it wasn't clear how we curate our `examples/` directory. If this api model works for your use case, please commit it somewhere canonical for yourself.

And if you feel like this is a real value-add generally (did you struggle to get this right?), then we'd welcome this with some comprehensive documentation that describes your scenario, and how you got acs-engine to work for it.

**Pull request closed without merging**